Source/Particles/Deposition/CurrentDeposition.H

```diff
@@ -251,27 +251,34 @@ void doEsirkepovDepositionShapeN (const Real * const xp, const Real * const yp,
                                   const Dim3 lo,
                                   const amrex::Real q)
 {
-
-#if (AMREX_SPACEDIM == 3)
-
     const Real dxi = 1.0/dx[0];
-    const Real dyi = 1.0/dx[1];
-    const Real dzi = 1.0/dx[2];
     const Real dtsdx0 = dt*dxi;
+    const Real xmin = xyzmin[0];
+#if (AMREX_SPACEDIM == 3)
+    const Real dyi = 1.0/dx[1];
     const Real dtsdy0 = dt*dyi;
+    const Real ymin = xyzmin[1];
+#endif
+    const Real dzi = 1.0/dx[2];
     const Real dtsdz0 = dt*dzi;
+    const Real zmin = xyzmin[2];
+
+#if (AMREX_SPACEDIM == 3)
     const Real invdtdx = 1.0/(dt*dx[1]*dx[2]);
     const Real invdtdy = 1.0/(dt*dx[0]*dx[2]);
     const Real invdtdz = 1.0/(dt*dx[0]*dx[1]);
+#elif (AMREX_SPACEDIM == 2)
+    const Real invdtdx = 1.0/(dt*dx[2]);
+    const Real invdtdz = 1.0/(dt*dx[0]);
+    const Real invvol = 1.0/(dx[0]*dx[2]);
+#endif
```

Comment:
Minor comment: `invvol` `invdtdx` `invdtdy` `invdtdz` are each used only once in this function. Should it be recomputed for each particle instead?

Reply:
Disregard this comment.